### PR TITLE
Default value

### PR DESCRIPTION
**Pull request by tbrand**
https://github.com/topaz-crystal/topaz/issues/8

### CODE REVIEW

File: src/topaz/model.cr [hunk 1]
```diff
@@ -28,15 +28,23 @@ module Topaz
...
         updated_at: Time?)
 
       def initialize({% for key, value in cols %}
                        {% if value.is_a?(NamedTupleLiteral) %}
                          {% if value[:nullable] %}
-                           @{{key.id}} : {{value[:type]}}?,
+                           {% if value[:default] %}
+                             @{{key.id}} : {{value[:type]}}? = {{value[:default]}},
+                           {% else %}
+                             @{{key.id}} : {{value[:type]}}?,
```
Comment (danielpclark) [inline src/topaz/model.cr]: Shouldn't this be @{{key.id}} : {{value[:type]}}? = nil,

File: src/topaz/model.cr [hunk 2]
```diff
@@ -278,9 +285,17 @@ module Topaz
             {% for key, value in cols %}
               {% if value.is_a?(NamedTupleLiteral) %}
                 {% if value[:nullable] %}
-                  {{key.id}} : {{value[:type]}}?,
+                  {% if value[:default] %}
+                    {{key.id}} : {{value[:type]}}? = {{value[:default]}},
+                  {% else %}
+                    {{key.id}} : {{value[:type]}}?,
```
Comment (danielpclark) [inline src/topaz/model.cr]: And this `{{key.id}} : {{value[:type]}}? = nil,`

File: src/topaz/model.cr [hunk 3]
```diff
@@ -296,9 +311,17 @@ module Topaz
...
```
Comment (danielpclark) [inline src/topaz/model.cr]: And lastly this is the same `{{key.id}} : {{value[:type]}}? = nil,`

File: src/topaz/model.cr (hunk 1 above)
Comment (danielpclark) [inline src/topaz/model.cr]: This is the `{% if value[:nullable] %}` path so I think this should be `= nil`

**Comment by danielpclark** [comment]
This looks good to me :+1:

If we want to change `nil` defaults that can be another issue/PR

**Comment by tbrand** [comment]
Sorry, I don't understand what you meant.
Just put `not_nil!` to returned values?

```
class Todo < Topaz::Model
  columns(
    description: String,
    completion: {type: Time, nullable: true} # `completion: Time?` is not allowed ...
  )
end

todo0 = Todo.new("Done", Time.now)
todo1 = Todo.new("Do something") # <- completion is nil

todo0.not_nil!.completion
todo1.not_nil!.completion # Crash!

if completion_time = todo1.completion # <- For safety
  puts "Done at #{completion_time}"
end
```

Basically union type is not allowed as column types since database column type have to be specific.
So `Time?` or `String|Int32` are not allowed. (But the compile will be completed successfully, that's a problem...)

Union of some specific type and Nil by using `{nullable: true}` is only allowed.

**Pull request merged**